lib/kafka/cluster.rb

```diff
@@ -420,30 +421,34 @@ def fetch_cluster_info
       errors = []
 
       @seed_brokers.shuffle.each do |node|
-        @logger.info "Fetching cluster metadata from #{node}"
-
-        begin
-          broker = @broker_pool.connect(node.hostname, node.port)
-          cluster_info = broker.fetch_metadata(topics: @target_topics)
-
-          if cluster_info.brokers.empty?
-            @logger.error "No brokers in cluster"
-          else
-            @logger.info "Discovered cluster metadata; nodes: #{cluster_info.brokers.join(', ')}"
-
-            @stale = false
-
-            return cluster_info
+        Resolv.getaddresses(node.hostname).select { |ip| ip =~ Resolv::IPv4::Regex }.shuffle.each do |ip|
```

Comment:
What about IPv6 addresses?

Reply:
I think currently ruby-kafka doesn't support IPv6 addresses because `Kafka::BrokerUri.parse` can't parse IPv6 addresses.

Reply:
I noticed that a hostname, a DNS record, could have a IPv6 address, and I didn’t think of such a case...

Reply:
Is there any reason to have this IPv4 regex at all? Why not simply:
```
Resolv.getaddresses(node.hostname).shuffle.each do |ip|
```

If an IPv6 address causes an exception further down the stack, better to solve that further down the stack, right? I don't actually follow what the issue would be with `Kafka::BrokerUri.parse` -- that's further up the stack and would be handling the text hostname before passing it down to this method.

If someone wanted to pass an IPv6 address in natively, the format should be like `kafka://[::1]:9092`

Reply:
I've changed the code as follows:

```
(@resolve_seed_brokers ? Resolv.getaddresses(node.hostname).shuffle : [node.hostname]).each
```

> I think currently ruby-kafka doesn't support IPv6 addresses because Kafka::BrokerUri.parse can't parse IPv6 addresses.

This is my misunderstanding. Although I don't remember well, I might have tried `Kafka.new("kafka://::1:9092")`...

> Is there any reason to have this IPv4 regex at all?

At first, I thought it was meaningless to try IPv6 addresses because both IPv4 addresses and IPv6 addresses might belong to the same broker. However we cannot know whether they belong to the same broker or not, and the DNS record might have only IPv6 addresses.
I also checked the behavior of librdkafka and found that it used both IPv4 addresses and IPv6 addresses even if they belonged to the same broker.
cf. https://github.com/edenhill/librdkafka/blob/v1.6.1/src/rdaddr.c#L162-L232

For the above reasons, I decided not to exclude IPv6 addresses.